subworkflows/nf-core/bam_baserecalibrator/main.nf

```diff
@@ -0,0 +1,59 @@
+//
+// PREPARE RECALIBRATION
+//
+
+include { GATK4_BASERECALIBRATOR  } from '../../../modules/nf-core/gatk4/baserecalibrator/main'
+include { GATK4_GATHERBQSRREPORTS } from '../../../modules/nf-core/gatk4/gatherbqsrreports/main'
+
+workflow BAM_BASERECALIBRATOR {
+    take:
+    ch_cram            // channel: [mandatory] [ meta, cram_markduplicates, crai ]
+    ch_dict            // channel: [mandatory] [ dict ]
+    ch_fasta           // channel: [mandatory] [ fasta ]
+    ch_fasta_fai       // channel: [mandatory] [ fasta_fai ]
+    ch_intervals       // channel: [mandatory] [ intervals, num_intervals ] (or [ [], 0 ] if no intervals)
+    ch_known_sites     // channel: [optional]  [ known_sites ]
+    ch_known_sites_tbi // channel: [optional]  [ known_sites_tbi ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Combine cram and intervals for spread and gather strategy
+    ch_cram_intervals = ch_cram.combine(ch_intervals)
+        // Move num_intervals to meta map
+        .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }
+
+    // RUN BASERECALIBRATOR
+    GATK4_BASERECALIBRATOR(
+        ch_cram_intervals,
+        ch_fasta.map{ it -> [ it ] },
```

Comment:
Can we update the module to take the metas?

Reply:
I am wondering if we should modify the input there aswell:
```
    input:
    tuple val(meta), path(input), path(input_index), path(intervals)
    path  fasta
    path  fai
    path  dict
    path  known_sites
    path  known_sites_tbi
```
and change it to
```
    input:
    tuple val(meta), path(input), path(input_index), path(intervals)
    tuple val(meta2), path(fasta), path(fai)
    tuple val(meta3), path(dict)
    tuple val(meta4), path(known_sites), path(known_sites_tbi)
```

Reply:
And if yes: Should I update it in a separate PR or can I do it here?